project_task_reference/tests/test_project_task.py

```diff
@@ -0,0 +1,77 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import pytest
+from ddt import ddt, data
+from odoo.exceptions import MissingError
+from odoo.tests import common
+from ..reference import TaskReference
+
+
+@ddt
+class TestProjectTask(common.SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.task_1 = cls.env['project.task'].create({'name': 'Task 1'})
+        cls.task_2 = cls.env['project.task'].create({'name': 'Task 2'})
+
+    @data(
+        'TA#{}',
+        'ta#{}',
+        'TA{}',
+        'ta{}',
+    )
+    def test_extract_references_from_text(self, reference_format):
+        reference = reference_format.format(self.task_1.id)
+        text = "Some prefix text {} some suffix text".format(reference)
+        result = self.env['project.task']._extract_references_from_text(text)
+        assert len(result) == 1
+        assert result[0].task_id == self.task_1.id
+
+    def test_reference_string(self):
+        expected_string = "ta#{}".format(self.task_1.id)
+        text = "Some prefix text {} some suffix text".format(expected_string)
+        reference = self.env['project.task']._extract_references_from_text(text)[0]
+        assert reference.string == expected_string
+
+    def test_reference_normalized_string(self):
+        text = "Some prefix text {} some suffix text".format("ta#{}".format(self.task_1.id))
+        reference = self.env['project.task']._extract_references_from_text(text)[0]
+        assert reference.normalized_string == "TA#{}".format(self.task_1.id)
+
+    def _make_reference_from_task_id(self, task_id):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4478774)